tests/ut/test_edge_directory_service.py

```diff
@@ -0,0 +1,137 @@
+        self._filer.get.assert_called_once_with('/config/fileservices/cifs/idMapping/map')
+        self._filer.put.assert_called_once_with('/config/fileservices/cifs/idMapping/map', mock.ANY)
+
+        expected_param = TestEdgeDirectoryService._get_advanced_mapping_object(self._domain_flat_name, self._mapping_min, self._mapping_max)
+        actual_param = self._filer.put.call_args[0][1]
+        self._assert_equal_objects(expected_param[0], actual_param[0])
+
+    def test_set_advanced_mapping_raise(self):
+        self.patch_call("cterasdk.edge.directoryservice.DirectoryService.domains")
+        get_response = self._get_advanced_mapping_object('Invalid domain name', 0, 0)
+        self._init_filer(get_response=get_response)
+        with self.assertRaises(exception.CTERAException) as error:
+            directoryservice.DirectoryService(self._filer).advanced_mapping(self._domain_flat_name, self._mapping_min, self._mapping_max)
+        self.assertEqual('Could not find domain name', error.exception.message)
+
+    def test_domains(self):
+        domain = Object()
+        domain.flatName = self._domain_flat_name
+        execute_response = [domain]
+        self._init_filer(execute_response=execute_response)
+        ret = directoryservice.DirectoryService(self._filer).domains()
+        self._filer.execute.assert_called_once_with('/status/fileservices/cifs', 'enumDiscoveredDomains')
+        self.assertEqual(ret[0], self._domain_flat_name)
+
+    def test_disconnect(self):
+        get_response = self._get_domain_param()
+        self._init_filer(get_response=get_response)
+        directoryservice.DirectoryService(self._filer).disconnect()
+        self._filer.get.assert_called_once_with('/config/fileservices/cifs')
+        self._filer.put.assert_called_once_with('/config/fileservices/cifs', mock.ANY)
+
+        expected_param = self._get_workgroup_param()
+        actual_param = self._filer.put.call_args[0][1]
+        self._assert_equal_objects(expected_param, actual_param)
+
+    @staticmethod
+    def _get_advanced_mapping_object(domain_flat_name, min_id, max_id):
+        mapping = Object()
+        mapping.domainFlatName = domain_flat_name
+        mapping.minID = min_id
+        mapping.maxID = max_id
+        return [mapping]
+
+    def _get_domain_join_param(self, ou=None):
+        o = Object()
+        o.username = self._username
+        o.password = self._password
+        if ou is not None:
+            o.ouPath = ou
+        return o
+
+    def _get_domain_param(self):
+        cifs_param = Object()
+        cifs_param.type = "domain"
+        cifs_param.workgroup = None
+        cifs_param.domain = self._domain
+        return cifs_param
+
+    def _get_workgroup_param(self):
+        cifs_param = Object()
+        cifs_param.type = "workgroup"
+        cifs_param.workgroup = self._workgroup
+        cifs_param.domain = None
+        return cifs_param
+
+    def _assert_equal_objects(self, expected_param, actual_param):
```

Comment:
Should be moved to the base class

Reply:
Thanks @ygalblum . Moved **_assert_equal_objects** to **base.py**